CircleciScripts/carthage-build.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Workaround for https://github.com/Carthage/Carthage/issues/3019
+# from https://github.com/Carthage/Carthage/issues/3019#issuecomment-665136323
+#
+# Update this to include new values as new Xcode 12 versions are released
+################################################################################
+
+# carthage-build.sh
+# Usage example: ./carthage-build.sh --platform iOS
+
+set -euo pipefail
+
+xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)
+trap 'rm -f "$xcconfig"' INT TERM HUP EXIT
+
+# For Xcode 12 make sure EXCLUDED_ARCHS is set to arm architectures otherwise
+# the build will fail on lipo due to duplicate architectures.
+# Xcode 12 Beta 3:
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A8169g = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+# Xcode 12 beta 4
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A8179i = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+# Xcode 12 beta 5
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A8189h = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+# Xcode 12 beta 6
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A8189n = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+# Xcode 12 GM (12A7208)
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A7208 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+# Xcode 12 GM (12A7209)
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_12A7209 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200 = $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200__BUILD_$(XCODE_PRODUCT_BUILD_VERSION))' >> $xcconfig
```

Comment:
Maybe it makes sense to check the presence of the variable and warn anyone using this script.. perhaps something like this:
```
if [[ -z "${XCODE_PRODUCT_BUILD_VERSION}" ]]; then
    echo "warning: XCODE_PRODUCT_BUILD_VERSION is not defined, carthage-build.sh needs to be updated to capture correct build version of xcode"
fi
```

(and maybe for the variable XCODE_VERSION_MAJOR as well?)

Reply:
Both `XCODE_PRODUCT_BUILD_VERSION` and `XCODE_VERSION_MAJOR` are populated by `xcodebuild`. Is your concern that they won't be present, or that we'll miss a future version of Xcode? If the latter, we should probably switch through the known product versions. In fact, we've already got a use case for this b/c Xcode 12.2 beta is already out

Reply:
Concern is if Xcode ever changes their variable names and then we start putting an empty string there without a clue on how to fix it.